docs/pages/guides/using-nextjs.mdx

```diff
@@ -6,123 +6,222 @@ description: A guide for integrating Next.js with Expo for the web.
+The package `react-native-web` builds on the assumption of reset CSS styles, here's how you reset styles in Next.js using the `pages/` directory.
 
-  - `touch next.config.js`
+```js pages/_document.js
+import { Children } from 'react';
+import Document, { Html, Head, Main, NextScript } from 'next/document';
+import { AppRegistry } from 'react-native';
 
-  **next.config.js**
+// Follows the setup for react-native-web:
+// https://necolas.github.io/react-native-web/docs/setup/#root-element
+// Plus additional React Native scroll and text parity styles for various
+// browsers.
+// Force Next-generated DOM elements to fill their parent's height
+const style = `
+html, body, #__next {
+  -webkit-overflow-scrolling: touch;
+}
+#__next {
+  display: flex;
+  flex-direction: column;
+  height: 100%;
+}
+html {
+  scroll-behavior: smooth;
+  -webkit-text-size-adjust: 100%;
+}
+body {
+  /* Allows you to scroll below the viewport; default value is visible */
+  overflow-y: auto;
+  overscroll-behavior-y: none;
+  text-rendering: optimizeLegibility;
+  -webkit-font-smoothing: antialiased;
+  -moz-osx-font-smoothing: grayscale;
+  -ms-overflow-style: scrollbar;
+}
+`;
+
+export default class MyDocument extends Document {
+  static async getInitialProps({ renderPage }) {
+    AppRegistry.registerComponent('main', () => Main);
+    const { getStyleElement } = AppRegistry.getApplication('main');
+    const page = await renderPage();
+    const styles = [
+      <style key="react-native-style" dangerouslySetInnerHTML={{ __html: style }} />,
+      getStyleElement(),
+    ];
+    return { ...page, styles: Children.toArray(styles) };
+  }
+
+  render() {
+    return (
+      <Html style={{ height: '100%' }}>
+        <Head />
+        <body style={{ height: '100%', overflow: 'hidden' }}>
+          <Main />
+          <NextScript />
+        </body>
+      </Html>
+    );
+  }
+}
+```
 
-  ```js
-  const { withExpo } = require('@expo/next-adapter');
-  const withPlugins = require('next-compose-plugins');
-  const withTM = require('next-transpile-modules')(['react-native-web']);
+```js pages/_app.js
+import Head from 'next/head';
+
+export default function App({ Component, pageProps }) {
+  return (
+    <>
+      <Head>
+        <meta name="viewport" content="width=device-width, initial-scale=1" />
+      </Head>
+      <Component {...pageProps} />
+    </>
+  );
+}
+```
 
-  const nextConfig = {};
+## Transpiling modules
 
-  module.exports = withPlugins([withTM, [withExpo, { projectRoot: __dirname }]], nextConfig);
-  ```
+By default, modules in the React Native ecosystem are not transpiled to run in web browsers. React Native relies on advanced caching in Metro to reload quickly. Next.js uses Webpack, which does not have the same level of caching, so by default, no node modules are transpiled. You will have to manually mark every module you want to transpile with the `transpilePackages` option in `next.config.js`:
```

Comment:
```suggestion
By default, modules in the React Native ecosystem are not transpiled to run in web browsers. React Native relies on advanced caching in Metro to reload quickly. Next.js uses Webpack, which does not have the same level of caching, so no node modules are transpiled by default. You will have to manually mark every module you want to transpile with the `transpilePackages` option in **next.config.js**:
```

[File names to be bold as per our writing styleguide](https://github.com/expo/expo/blob/main/guides/Expo%20Documentation%20Writing%20Style%20Guide.md#filenames-as-bold-text).

Also, we are repeating the context of "by default" twice in the paragraph above. Kept the first instance and remove the second one.